src/coreclr/build-runtime.sh

```diff
@@ -145,8 +145,11 @@ export MSBUILDDEBUGPATH
 # Check prereqs.
 check_prereqs
 
+# When the host runs on an unknown rid, it falls back to the output rid
+__FallbackOS="${__OutputRid%-*}" # Strip architecture
```

Comment:
nit: move this assignment after https://github.com/dotnet/runtime/blob/3a3692029a2a038581fdc8c1c6496e0506a7e0c7/eng/native/build-commons.sh#L492 and then we won't need to repeat this logic in other scripts.